bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkBuilderImpl.java

```diff
@@ -294,4 +305,63 @@ private Pair<Page, String> resolvePage(@NotNull final Page page) {
         return new ImmutablePair<>(resolved, getPageLinkURL(resolved));
     }
 
+    /**
+     * Masks a given {@link String} by replacing all occurrences of {@link LinkBuilderImpl#PATTERNS} with a placeholder.
+     * The generated placeholders are put into the given {@link Map} and can be used to unmask a {@link String} later on.
+     *
```

Comment:
Could you please add an example to help understand what this method does, e.g. with `/content/path/to/page.html?recipient=<%= recipient.id %>`?